src/App.tsx

```diff
@@ -1,26 +1,423 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useRef, useState, useMemo } from 'react';
+import classNames from 'classnames';
 import { UserWarning } from './UserWarning';
+import { getTodos, addTodo, deleteTodo, updateTodo } from './api/todos';
+import { Todo } from './types/Todo';
+import { USER_ID } from './utils/preferences';
 
-const USER_ID = 0;
+export type Filter = 'all' | 'active' | 'completed';
+
+enum ErrorMessage {
+  Load = 'Unable to load todos',
+  Empty = 'Title should not be empty',
+  Add = 'Unable to add a todo',
+  Delete = 'Unable to delete a todo',
+  Update = 'Unable to update a todo',
+}
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [errorMessage, setErrorMessage] = useState('');
```

Comment:
where is type?
you should use enum here
```suggestion
  const [errorMessage, setErrorMessage] = useState(ErrorMessage.Default);
```

Reply:
fixed it, thank you!